app/services/candidates/registrations/behaviours/placement_preference.rb

```diff
@@ -8,14 +8,24 @@ module PlacementPreference
         MAX_WORDS_FOR_OBJECTIVE = 149
 
         included do
-          validates :availability, presence: true
+          validates :urn, presence: true
+          validates :bookings_placement_date_id,
+            presence: true,
+            if: -> { urn.present? && school_offers_fixed_dates? }
```

Comment:
Hopefully when I get round to refactoring how these candidate registration steps work we can remove this urn presence check as it's not super obvious why this needs to be done.